src/tests/end2end/RenderPassLoadOpTests.cpp

```diff
@@ -0,0 +1,179 @@
+                .SetFormat(nxt::TextureFormat::R8G8B8A8Unorm)
+                .SetMipLevels(1)
+                .SetAllowedUsage(nxt::TextureUsageBit::OutputAttachment | nxt::TextureUsageBit::TransferSrc)
+                .SetInitialUsage(nxt::TextureUsageBit::OutputAttachment)
+                .GetResult();
+
+            renderTargetView = renderTarget.CreateTextureViewBuilder().GetResult();
+
+            RGBA8 zero(0, 0, 0, 0);
+            std::fill(expectZero.begin(), expectZero.end(), zero);
+
+            RGBA8 green(0, 255, 0, 255);
+            std::fill(expectGreen.begin(), expectGreen.end(), green);
+
+            RGBA8 blue(0, 0, 255, 255);
+            std::fill(expectBlue.begin(), expectBlue.end(), blue);
+
+            // draws a blue quad on the right half of the screen
+            const char* vsSource = R"(
+                #version 450
+                void main() {
+                    const vec2 pos[6] = vec2[6](
+                        vec2(0, -1), vec2(1, -1), vec2(0, 1),
+                        vec2(0,  1), vec2(1, -1), vec2(1, 1));
+                    gl_Position = vec4(pos[gl_VertexIndex], 0.f, 1.f);
+                }
+                )";
+            const char* fsSource = R"(
+                #version 450
+                out vec4 color;
+                void main() {
+                    color = vec4(0.f, 0.f, 1.f, 1.f);
+                }
+                )";
+            blueQuad = DrawQuad(&device, vsSource, fsSource);
+        }
+
+        nxt::Texture renderTarget;
+        nxt::TextureView renderTargetView;
+
+        std::array<RGBA8, kRTSize * kRTSize> expectZero;
+        std::array<RGBA8, kRTSize * kRTSize> expectGreen;
+        std::array<RGBA8, kRTSize * kRTSize> expectBlue;
+
+        DrawQuad blueQuad = {};
+};
+
+TEST_P(RenderPassLoadOpTests, ClearOnceLoadOnce) {
+    // Part 1: clear once, check to make sure it's cleared
+
+    auto renderpass1 = device.CreateRenderPassBuilder()
+        .SetAttachmentCount(1)
+        .SetSubpassCount(1)
+        .AttachmentSetFormat(0, nxt::TextureFormat::R8G8B8A8Unorm)
+        .AttachmentSetColorLoadOp(0, nxt::LoadOp::Clear)
+        .SubpassSetColorAttachment(0, 0, 0)
+        .GetResult();
+    auto framebuffer1 = device.CreateFramebufferBuilder()
+        .SetRenderPass(renderpass1)
+        .SetDimensions(kRTSize, kRTSize)
+        .SetAttachment(0, renderTargetView)
+        .GetResult();
+    framebuffer1.AttachmentSetClearColor(0, 0.0f, 1.0f, 0.0f, 1.0f); // green
+
+    auto commands1 = device.CreateCommandBufferBuilder()
+        .BeginRenderPass(renderpass1, framebuffer1)
+        .BeginRenderSubpass()
+            // Clear should occur implicitly
+            // Store should occur implicitly
+        .EndRenderSubpass()
+        .EndRenderPass()
+        .GetResult();
+
+    // Initialized to 0 before the command buffer is submitted
+    EXPECT_TEXTURE_RGBA8_EQ(expectZero.data(), renderTarget, 0, 0, kRTSize, kRTSize, 0);
```

Comment:
This isn't true yet on any backend, and would be something enabled by browsers when creating the device. It could be replaced with a first clear to black, then a clear to green.